File: graphql-java-kickstart-webclient/src/main/java/com/example/demo/gql/types/Author.java
package com.example.demo.gql.types;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Author {
  private String id;

  private String name;

  private String email;

  private LocalDateTime createdAt;

  @Builder.Default
  private List<Post> posts = new ArrayList<>();

}

File: graphql-kotlin/src/main/kotlin/com/example/demo/gql/types/Author.kt
package com.example.demo.gql.types

import java.time.LocalDateTime
import java.util.*

data class Author(
    val id: UUID?,
    val name: String,
    val email: String,
    val createdAt: LocalDateTime? = null,
    val posts: List<Post>? = emptyList()
)